WordPress/Classes/Utility/NotificationsManager.m

```diff
@@ -56,23 +56,15 @@ + (void)registerForPushNotifications
 #if TARGET_IPHONE_SIMULATOR || ALPHA_BUILD
     return;
 #endif
+    // iOS 8 or higher notifications registration
+    [[UIApplication sharedApplication] registerForRemoteNotifications];
 
-    BOOL canRegisterUserNotifications = [[UIApplication sharedApplication] respondsToSelector:@selector(registerForRemoteNotifications)];
-    if (!canRegisterUserNotifications) {
-        // iOS 7 notifications registration
-        UIRemoteNotificationType types = (UIRemoteNotificationTypeBadge | UIRemoteNotificationTypeSound | UIRemoteNotificationTypeAlert);
-        [[UIApplication sharedApplication] registerForRemoteNotificationTypes:types];
-    } else {
-        // iOS 8 or higher notifications registration
-        [[UIApplication sharedApplication] registerForRemoteNotifications];
-
-        // Add the categories to UIUserNotificationSettings
-        UIUserNotificationType types = UIUserNotificationTypeBadge | UIUserNotificationTypeSound | UIUserNotificationTypeAlert;
-        UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:types categories:[self buildNotificationCategories]];
-
-        // Finally, register the notification settings
-        [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
-    }
```

Comment:
(Thank you for killing this!). I'm in the process of rewriting this whole class (#4562)